packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -956,19 +956,71 @@ const alternateTypeKey = createStateSymbol("alternateType");
 export const $alternateType: AlternateTypeDecorator = (
   context: DecoratorContext,
   source: ModelProperty | Scalar,
-  alternate: Scalar,
+  alternate: Type,
   scope?: LanguageScopes,
 ) => {
-  if (source.kind === "ModelProperty" && source.type.kind !== "Scalar") {
+  let invalidAlternate = true;
+  let invalidSource = true;
+
+  // if alternate type is unknown, source type must be scalar
```

Comment:
i'm wondering why this validation is needed? change to `unknown` could apply to any type in my opinion.

Reply:
Per my understanding, the reason why we introduce the `unknown` type is to serialize some Scalar value into binary data. So I add the validation here to limit the usage of `unknown`. @pshao25 @ArcturusZhang @live1206  could you help to put your thoughts regarding whether we need this validation here?

Reply:
The validation does look odd with all these limitations.
As offline discussed, we would like any/unknown for target of TypeProperty.
But for target of `Scalar`, we only allow `Scalar`.

Reply:
@live1206 @tadelesh the unnecessary type validation logic has been removed, please help to review again.